### PR TITLE
Mkl fix pr (fix an issue in prepare_mkl.sh and add test case)

### PR DESCRIPTION
**Pull request by jinhuang415**
## Description ##
Submit pull request for an issue in prepare_mkl.sh (more details please see https://github.com/apache/incubator-mxnet/issues/9112)

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: tests/python/cpu/test_mklml.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import os
+
+def test_mklml_install():
+    """
+    This function will check if MXNet is built/installed correctly
+    when compiling with Intel MKLML library, the method is try
+    to import mxnet module and see if correct mklml library is
+    mapped to this process's address space
+    """
+    logging.basicConfig(level=logging.INFO)
```
Comment (KellenSunderland) [inline tests/python/cpu/test_mklml.py]: Will this change the global logging level?  Would this be a problem if say a user wanted to run the entire test suit in warning only mode?

File: tests/python/cpu/test_mklml.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    try:
+        #pylint: disable=unused-variable
+        import mxnet as mx
+    except (ImportError, OSError) as e:
+        assert 0, "Import mxnet error: %s. Please double check your build/" \
+               "install steps or environment variable settings" % str(e)
+
+    pid = os.getpid()
+    rc = os.system("cat /proc/" + str(pid) + \
```
Comment (KellenSunderland) [inline tests/python/cpu/test_mklml.py]: Does this work on windows?  If not, would it be possible to detect if you're running in Windows and skip the test?

File: tests/python/cpu/test_mklml.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import logging
+import os
+
+def test_mklml_install():
+    """
+    This function will check if MXNet is built/installed correctly
+    when compiling with Intel MKLML library, the method is try
+    to import mxnet module and see if correct mklml library is
+    mapped to this process's address space
```
Comment (KellenSunderland) [inline tests/python/cpu/test_mklml.py]: Copy suggestion: This test will verify that MXNet is built/installed correctly when compiled with Intel MKLML library.  The method will try to import the mxnet module and see if the mklml library is mapped to this process's address space.

File: tests/python/cpu/test_mklml.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+"""
+MKLML related test cases
+"""
+
+import logging
+import os
+
+def test_mklml_install():
+    """
+    This function will check if MXNet is built/installed correctly
```
Comment (KellenSunderland) [inline tests/python/cpu/test_mklml.py]: Copy suggestion: This test will verify that MXNet is built/installed correctly when compiled with Intel MKLML library.  The method will try to import the mxnet module and see if the mklml library is mapped to this process's address space.

Comment (jinhuang415) [reply tests/python/cpu/test_mklml.py]: Thanks for the suggestion, will update the docstring accordingly.

**Comment by jinhuang415** [comment]
My previous thought is to run this test by hand when we want to verify if MKLML is installed correctly, and this test case only applies to Linux (not applied to windows/MAC and other OS) so if needed I can add check for Linux OS and skip check for other OS. This test case will pass only when installed with MKLML, so if using openblas or GPU it will fail. So if we want to add it into the Jenkins we may only test with MKLML scenario. I checked Jenkinsfile, in order to add this into CI build, looks like we need to add new def like python2[3]_mklml_ut() to only invoke my newly added test case (tests/python/cpu/test_mklml.py) and invoked by MKLML build only, doing this can make sure all MKLML build linked with correct library for sanity check, but it will also bring the Jenkins logic a little more complicated.
Please let me know your suggestions on it.

**Comment by marcoabreu** [comment]
LGTM

**Pull request merged**